### PR TITLE
[ICD] Initial Client Monitoring Cluster xml update

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Description
PR updates the client monitoring cluster xml to match the cm cluster spec update 
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/6482

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 1]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
-        <description>Client Monitoring allows for ensuring that listed clients meet the required monitoring conditions on the server.</description>
-        <attribute side="server" code="0x00" define="IDLE_MODE_INTERVAL" type="INT32U" min="0x12C" max="0X5265C00" default="0x12C" writable="false" optional="false" isNullable="false">IdleModeInterval</attribute>
-        <attribute side="server" code="0x01" define="ACTIVE_MODE_INTERVAL" type="INT32U" min="0x12C" max="0X5265C00" default="0x12C" writable="false" optional="false" isNullable="false">ActiveModeInterval</attribute>
-        <attribute side="server" code="0x02" define="ACTIVE_MODE_THRESHOLD" type="INT16U" min="0x12C" max="0xEA60" default="0xFA0" writable="false" optional="false" isNullable="false">ActiveModeThreshold</attribute>
+        <description>Client Monitoring allows for ensuring that listed clients are notified when a server is available for communication.</description>
+        <attribute side="server" code="0x00" define="IDLE_MODE_INTERVAL" type="INT32U" min="0x1F4" max="0X5265C00" default="0x1F4" writable="false" optional="false" isNullable="false">IdleModeInterval</attribute>
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml]: yep; done

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 2]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
 <configurator>
     <domain name="CHIP"/>
 
-    <struct name = "MonitoringRegistration" isFabricScoped="true">
+    <struct name = "MonitoringRegistrationStruct" isFabricScoped="true">
         <cluster code="0x1046"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:22]: Where did this cluster code come from?  The spec does not have a cluster code assigned yet, which we should fix ASAP.

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 3]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
         <item fieldId="1" name="ClientNodeId" type="node_id"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:23]:
```suggestion
        <item fieldId="1" name="ClientNodeID" type="node_id" isFabricSensitive="true"/>
```

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 4]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
-        <item fieldId="2" name="ICid" type="INT64U"/>
+        <item fieldId="2" name="Key" type="OCTET_STRING" length="16"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:24]:
```suggestion
        <item fieldId="2" name="Key" type="OCTET_STRING" length="16" isFabricSensitive="true"/>
```

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 5]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
     <cluster>
         <domain>General</domain>
         <name>Client Monitoring</name>
         <code>0x1046</code>
         <define>CLIENT_MONITORING_CLUSTER</define>
-        <description>Client Monitoring allows for ensuring that listed clients meet the required monitoring conditions on the server.</description>
-        <attribute side="server" code="0x00" define="IDLE_MODE_INTERVAL" type="INT32U" min="0x12C" max="0X5265C00" default="0x12C" writable="false" optional="false" isNullable="false">IdleModeInterval</attribute>
-        <attribute side="server" code="0x01" define="ACTIVE_MODE_INTERVAL" type="INT32U" min="0x12C" max="0X5265C00" default="0x12C" writable="false" optional="false" isNullable="false">ActiveModeInterval</attribute>
-        <attribute side="server" code="0x02" define="ACTIVE_MODE_THRESHOLD" type="INT16U" min="0x12C" max="0xEA60" default="0xFA0" writable="false" optional="false" isNullable="false">ActiveModeThreshold</attribute>
+        <description>Client Monitoring allows for ensuring that listed clients are notified when a server is available for communication.</description>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:32]:
```suggestion
        <description>Client Monitoring allows servers to ensure that listed clients are notified when a server is available for communication.</description>
```

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 6]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
+        <attribute side="server" code="0x00" define="IDLE_MODE_INTERVAL" type="INT32U" min="500" max="86400000" default="500" writable="false" optional="false" isNullable="false">IdleModeInterval</attribute>
+        <attribute side="server" code="0x01" define="ACTIVE_MODE_INTERVAL" type="INT32U" min="300" default="300" writable="false" optional="false" isNullable="false">ActiveModeInterval</attribute>
+        <attribute side="server" code="0x02" define="ACTIVE_MODE_THRESHOLD" type="INT16U" min="300" default="300" writable="false" optional="false" isNullable="false">ActiveModeThreshold</attribute>
 
-        <!-- TODO: Revisit the number of entries per fabric-->
-        <attribute side="server" code="0x03" define="EXPECTED_CLIENTS" type="ARRAY" entryType="MonitoringRegistration" length="4" writable="false" optional="false" isNullable="false">ExpectedClients</attribute>
+        <attribute side="server" code="0x03" define="EXPECTED_CLIENTS" type="ARRAY" entryType="MonitoringRegistrationStruct" writable="false" optional="false" isNullable="false" reportable="false">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:37]:
```suggestion
        <attribute side="server" code="0x03" define="EXPECTED_CLIENTS" type="ARRAY" entryType="MonitoringRegistrationStruct" writable="false" optional="false" isNullable="false">
```

"reportable" in this XML has nothing to do with Matter.

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 7]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
+            <description>ExpectedClients</description>
+            <access op="read" privilege="administer"/>
+            <access modifier="fabric-scoped"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:40]: I'm not sure this `<access modifier> bit actually does anything, fwiw, but leaving it is OK for now.  Please file an issue to figure out whether it does do something.

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 8]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
+        </attribute>
 
-        <command source="client" code="0x00" name="RegisterClientMonitoring" optional="false">
+        <attribute side="server" code="0x04" define="ICD_COUNTER" type="INT32U" writable="false" optional="false" isNullable="false" reportable="false">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:43]:
```suggestion
        <attribute side="server" code="0x04" define="ICD_COUNTER" type="INT32U" writable="false" optional="false" isNullable="false">
```
Again, "reportable" is not related to Matter...  we do need a way to represent "C" in XML, but we don't have one right now.

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 9]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
+            <description>ICDCounter</description>
+            <access op="read" privilege="administer"/>
+        </attribute>
+
+
+        <command source="client" code="0x00" name="RegisterClientMonitoring" response="RegisterClientMonitoringResponse" isFabricScoped="true" optional="false">
             <description> Register a client to the end device </description>
             <arg name="ClientNodeId" type="node_id"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:51]:
```suggestion
            <arg name="ClientNodeID" type="node_id"/>
```

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 10]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
-            <arg name="ICid" type="INT64U"/>
+            <arg name="Key" type="OCTET_STRING" length="16"/>
             <access op="invoke" privilege="manage"/>
         </command>
 
-        <command source="client" code="0x01" name="UnregisterClientMonitoring" optional="false">
+        <command source="server" code="0x01" name="RegisterClientMonitoringResponse" optional="false" disableDefaultResponse="true">
+            <description> RegisterClientMonitoring response command </description>
+            <arg name="Status" type="ENUM8"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:58]: Can you please file a followup to make the `status` type actually work right and use it here?  Feel free to assign it to me.

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 11]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
+            <arg name="ICDCounter" type="INT32U" isNullable="true"/>
+        </command>
+
+        <command source="client" code="0x02" name="UnregisterClientMonitoring" isFabricScoped="true" optional="false">
             <description> Unregister a client from an end device </description>
             <arg name="ClientNodeId" type="node_id"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:64]:
```suggestion
            <arg name="ClientNodeID" type="node_id"/>
```

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml (hunk 2 above)
Comment (mkardous-silabs) [reply src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:22]: Random one i had chosen when i first did the cluster xml. There is a ticket assigned to @robszewczyk for this.
https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/5903

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml (hunk 7 above)
Comment (mkardous-silabs) [reply src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:40]: Opened #26316

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml (hunk 10 above)
Comment (mkardous-silabs) [reply src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:58]: opened #26317

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml [hunk 12]
```diff
@@ -18,40 +18,54 @@ limitations under the License.
...
+        <attribute side="server" code="0x01" define="ACTIVE_MODE_INTERVAL" type="INT32U" min="300" default="300" writable="false" optional="false" isNullable="false">ActiveModeInterval</attribute>
+        <attribute side="server" code="0x02" define="ACTIVE_MODE_THRESHOLD" type="INT16U" min="300" default="300" writable="false" optional="false" isNullable="false">ActiveModeThreshold</attribute>
 
-        <!-- TODO: Revisit the number of entries per fabric-->
-        <attribute side="server" code="0x03" define="EXPECTED_CLIENTS" type="ARRAY" entryType="MonitoringRegistration" length="4" writable="false" optional="false" isNullable="false">ExpectedClients</attribute>
+        <attribute side="server" code="0x03" define="EXPECTED_CLIENTS" type="ARRAY" entryType="MonitoringRegistrationStruct" writable="false" optional="false" isNullable="false">
+            <description>ExpectedClients</description>
+            <access op="read" privilege="administer"/>
+            <access modifier="fabric-scoped"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml:40]: `<access modifier>` does nothing.  Please take it out.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml (hunk 12 above)
Comment (mkardous-silabs) [reply src/app/zap-templates/zcl/data-model/chip/client-monitoring-cluster.xml]: done

**Pull request merged**